android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BeaconConfigFragment.java

```diff
@@ -327,21 +343,33 @@ private void saveEditCardUrlToBeacon() {
     if (!URLUtil.isNetworkUrl(url)) {
       url = "http://" + url;
     }
+    // Create the callback object to set the url
+    UrlShortener.ModifiedUrlCallback urlSetter = new UrlShortener.ModifiedUrlCallback() {
+      @Override
+      public void onNewUrl(String newUrl){
+        // Write the url to the device
+        try {
```

Comment:
You should abstract this and call it `setUriBeaconUrl` and use it here and in line 372.